vividus-plugin-json/src/main/java/org/vividus/json/steps/JsonNestedSteps.java

```diff
@@ -0,0 +1,276 @@
+     * expected value. No steps will be performed in case of comparison rule mismatch</li>
+     * <li>Restores previously set context</li>
+     * <li>Step will fail if variable never be set along the iterations</li>
+     * </ul>
+     * <br> Usage example:
+     * <code>
+     * <br>When I find > `1` JSON elements from `${json}` by `$.store.book` and until variable `title` matches `M.+`
+     * for each element do:
+     * <br>|step|
+     * <br>|When I save JSON element value from context by JSON path `$.title` to scenario variable `title`|
+     * </code>
+     * @param comparisonRule  The rule to match the quantity of elements. The supported rules:
+     *                        <ul>
+     *                        <li>less than (&lt;)</li>
+     *                        <li>less than or equal to (&lt;=)</li>
+     *                        <li>greater than (&gt;)</li>
+     *                        <li>greater than or equal to (&gt;=)</li>
+     *                        <li>equal to (=)</li>
+     *                        <li>not equal to (!=)</li>
+     *                        </ul>
+     * @param elementsNumber  The expected number of elements
+     * @param json            A JSON element
+     * @param jsonPath        A JSON path
+     * @param variableName    A variable name to check
+     * @param variableMatcher The rule to match the quantity of elements. The supported rules:
+     *                        <ul>
+     *                        <li>is equal to</li>
+     *                        <li>contains</li>
+     *                        <li>does not contain</li>
+     *                        <li>matches</li>
+     *                        </ul>
+     * @param expectedValue   The expected variable value
+     * @param stepsToExecute  Examples table with steps to execute for each found elements
+     */
+    @When("I find $comparisonRule `$elementsNumber` JSON elements from `$json` by `$jsonPath`"
+            + " and until variable `$variableName` $variableMatcher `$expectedValue`"
+            + " for each element do:$stepsToExecute")
+    @Alias("I find $comparisonRule '$elementsNumber' JSON elements from '$json' by '$jsonPath' and"
+            + " until variable '$variableName' $variableMatcher '$expectedValue' for each element do:$stepsToExecute")
+    public void performAllStepsForJsonEntriesExpectingVariable(ComparisonRule comparisonRule, int elementsNumber,
+        String json, String jsonPath, String variableName, StringComparisonRule variableMatcher, String expectedValue,
+            SubSteps stepsToExecute)
+    {
+        Matcher<String> matcher = variableMatcher.createMatcher(expectedValue);
+        performStepsForEachJsonEntry(json, jsonPath, comparisonRule, elementsNumber, stepsToExecute, () -> {
+            Object variable = variableContext.getVariable(variableName);
+            return variable != null && matcher.matches(String.valueOf(variable));
```

Comment:
double check on null? :)

Reply:
Please elaborate.

Reply:
1) variable != null
2) second inside String.valueOf
maybe just variable.toString instead of String.valueOf(variable)